src/mongocxx/private/database.hpp

```diff
@@ -34,6 +34,19 @@ class database::impl {
         name(std::move(name))
     {}
 
+    impl(const impl& i) : 
+        database_t{libmongoc::database_copy(i.database_t)},
+        client_impl{i.client_impl},
+        name{i.name} 
+    {}
+
+    impl& operator=(const impl& i) {
+        database_t = libmongoc::database_copy(i.database_t);
```

Comment:
not here, but the destructor is called in the copy assignment operator of mongocxx::database (which is what actually matters) since we don't actually use the impl's copy assignment operator there.